### PR TITLE
Feature/40199 spa connection check config fixes

### PR DESCRIPTION
**Pull request by GuyHarwood**
- allow sourcing from global `.env` file at solution root
- add clarity around sourcing of azure storage implementation
- format example.env.txt into service areas

### CODE REVIEW

File: pupil-spa/gen_config.sh [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+# support root .env file for local development scenarios...
+if [ -f ../.env ]
+then
+  # enables all following variable definitions to be exported...
+  set -o allexport
+  source ../.env
+  # disable it again
+  set +o allexport
+fi
+
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/gen_config.sh]: really nice to see this

File: pupil-spa/gen_config.sh [hunk 2]
```diff
@@ -1,4 +1,15 @@
 #!/bin/bash
+
+# support root .env file for local development scenarios...
+if [ -f ../.env ]
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply pupil-spa/gen_config.sh]: updated and tested, thanks for suggestion

File: pupil-spa/src/app/services/azure-queue/azure-queue.service.ts [hunk 3]
```diff
@@ -54,19 +57,33 @@ export class AzureQueueService {
    * @param {Object} retryConfig
    * @returns {Promise.<Object>}
    */
-  public async addMessage(queueName: string, url: string, token: string, payload: object, retryConfig: object): Promise<Object> {
+  public async addMessage(queueName: string, url: string, token: string, payload: object,
+     retryConfig: IRetryConfig, serviceProperties?: IQueueServiceProperties): Promise<Object> {
     const queueService = this.initQueueService(queueName, url, token, retryConfig);
+    if (serviceProperties) {
+      console.log(`GUY: setting the service props:${JSON.stringify(serviceProperties, null, 2)}`)
```
Comment (jon-shipley) [inline pupil-spa/src/app/services/azure-queue/azure-queue.service.ts:64]: Debug?

File: pupil-spa/src/app/services/azure-queue/azure-queue.service.ts [hunk 4]
```diff
@@ -54,19 +57,33 @@ export class AzureQueueService {
...
+      const result = await queueService.setServiceProperties(serviceProperties);
+      console.log(`GUY: result of set props:${JSON.stringify(result, null, 2)}`)
```
Comment (jon-shipley) [inline pupil-spa/src/app/services/azure-queue/azure-queue.service.ts:66]: Debug?

**Comment by GuyHarwood** [comment]
Queue storage ping request must be rewritten to use basic request object due to constraints with MS library

**Pull request closed without merging**